ui-admin/src/util/RequireUserPermission.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+import {
+  userHasPermission,
+  useUser
+} from 'user/UserProvider'
+import { Portal } from '@juniper/ui-core'
+
+
+export type RequireUserPermissionProps = RequireSuperuser | RequirePermissions
+
+type BaseRequireUserPermissionProps = {
+  noPermissionMessage?: React.ReactNode,
+  children: React.ReactNode,
+}
+
+type RequireSuperuser = BaseRequireUserPermissionProps & {
+  superuser: true
+}
+
+const isRequireSuperuser = (props: RequireUserPermissionProps): props is RequireSuperuser => {
+  return (props as RequireSuperuser).superuser
+}
+
+type RequirePermissions = BaseRequireUserPermissionProps &  {
+  portal: Portal,
```

Comment:
how about making the portal argument optional, and using 
`const { portal } = useContext(PortalContext) as PortalContextT` to get the portal by default?

Reply:
atually, nevermind -- don't do that -- we need to use this component outside of portal contexts sometimes for showing the superuser sidebar.  Although, that might be an argument that RequireSuperuser and RequirePortalPermission should be separate components